File: content/examples/other/leetcode/problems/best-time-to-buy-and-sell-stock/example-1/Solution.java
public class Solution {
  public static int maxProfit(int[] prices) {
    int best = 0;
    for (int i = 0; i < prices.length; i++) {
      for (int j = i + 1; j < prices.length; j++) {
        best = Math.max(best, prices[j] - prices[i]);
      }
    }
    return best;
  }

  public static void main(String[] args) {
    System.out.println(maxProfit(new int[]{7,1,5,3,6,4}));
  }
}

File: content/examples/other/leetcode/problems/best-time-to-buy-and-sell-stock/example-1/solution.js
function maxProfit(prices) {
  let best = 0;
  for (let i = 0; i < prices.length; i += 1) {
    for (let j = i + 1; j < prices.length; j += 1) {
      best = Math.max(best, prices[j] - prices[i]);
    }
  }
  return best;
}

if (require.main === module) {
  console.log(maxProfit([7,1,5,3,6,4]));
}

module.exports = { maxProfit };
